flow/project.py

```diff
@@ -850,37 +848,13 @@ def _determine_entrypoint(self, entrypoint, directives, jobs):
         self._set_entrypoint_item(entrypoint, directives, 'path', default_path, jobs)
         return "{} {}".format(entrypoint['executable'], entrypoint['path']).lstrip()
 
-    def _resolve_directives(self, name, defaults, jobs):
+    def _resolve_directives(self, name, defaults, env):
+        base_directives = env._get_default_directives()
```

Comment:
I don't like the name `base_directives` because it's being updated and returned (and is no longer just the "base" directives). A name like `directives` or `all_directives` might be better. (I can't make a code suggestion to help here because of how the diff is structured.)